examples/docs_snippets/docs_snippets/concepts/resources/pythonic_resources.py

```diff
@@ -580,3 +580,128 @@ def public_github_repos(github: Resource[GitHub]):
     )
 
     # end_raw_github_resource_factory
+
+
+def new_resource_on_sensor() -> None:
+    # start_new_resource_on_sensor
+    from dagster import (
+        sensor,
+        RunRequest,
+        SensorEvaluationContext,
+        ConfigurableResource,
+        job,
+        Definitions,
+        RunConfig,
+    )
+    import requests
+    from typing import List
+
+    class UsersAPI(ConfigurableResource):
+        url: str
+
+        def fetch_users(self) -> List[str]:
+            return requests.get(self.url).json()
+
+    @job
+    def process_user():
+        ...
+
+    @sensor(job=process_user)
+    def process_new_users_sensor(
+        context: SensorEvaluationContext,
+        users_api: UsersAPI,
+    ):
+        last_user = int(context.cursor) if context.cursor else 0
+        users = users_api.fetch_users()
+
+        num_users = len(users)
+        for user_id in users[last_user:]:
+            yield RunRequest(
+                run_key=user_id,
+                tags={"user_id": user_id},
+            )
+
+        context.update_cursor(str(num_users))
+
+    defs = Definitions(
+        jobs=[process_user],
+        sensors=[process_new_users_sensor],
+        resources={"users_api": UsersAPI(url="https://my-api.com/users")},
+    )
+    # end_new_resource_on_sensor
+
+    # start_test_resource_on_sensor
+
+    from dagster import build_sensor_context, validate_run_config
+
+    def test_process_new_users_sensor():
+        class MockUsersAPI(ConfigurableResource):
+            def fetch_users(self) -> List[str]:
+                return ["1", "2", "3"]
+
+        context = build_sensor_context()
+        run_requests = process_new_users_sensor(context, users_api=MockUsersAPI())
```

Comment:
I assume this does not pass type checking?

Reply:
It's not type checked since the type signature is a generic sensor evaluation type function, which takes generic args/kwargs.

Reply:
Why have it inherit from ConfigurableResource at all then?

Reply:
good point - dropped